ivy/functional/frontends/torch/tensor.py

```diff
@@ -281,6 +281,9 @@ def max(self, dim=None, keepdim=False):
     def device(self):
         return ivy.dev(self.data)
 
+    def is_cuda(self):
+        return self.device == "gpu"
```

Comment:
I think you should use the ivy.dev(self.data) used in the pervious instance function to get the device where the data is stored.

Reply:
Also, the device is of a special datatype in Ivy, check this document for details: https://lets-unify.ai/ivy/deep_dive/devices.html

Reply:
Right now, I am going about in this way:

From @given, I am taking bool values for device, then if device is True, I use "gpu:0" else "cpu" inside ivy.Device

Reply:
Is this the correct way to proceed?

Reply:
I think so, maybe you can allow all three possible device (cpu

gpu:idx

tpu:idx) to be generated?